clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -747,21 +748,19 @@ public void testOffsetCommitSyncTimeoutNotReturnedOnPollAndFails() {
             new TopicPartition("topic", 1),
             new OffsetAndMetadata(0));
 
-        // Send sync offset commit request that fails with retriable error.
-        long expirationTimeMs = time.milliseconds() + retryBackoffMs * 2;
-        CompletableFuture<Void> commitResult = commitRequestManager.commitSync(offsets, expirationTimeMs);
-        completeOffsetCommitRequestWithError(commitRequestManager, Errors.REQUEST_TIMED_OUT);
+        // Send sync offset commit request.
+        long deadlineMs = time.milliseconds() + retryBackoffMs * 2;
+        CompletableFuture<Void> commitResult = commitRequestManager.commitSync(offsets, deadlineMs);
```

Comment:
This logic needed to be revised as the previous code relied on the preemptive pruning of expired requests.